### ISSUE TITLE
Support quoted strings as js_name values

### ISSUE THREAD

**Issue by brisad** [rustwasm/wasm-bindgen#583]
In addition to supporting `js_name = foo` we should also support a quoted variant like `js_name = "$1"`. This is needed to be able to create bindings for the methods `$1-$9` in `RegExp`. (#275)

**Comment by brisad** [comment]
I'm interested in working on this, unless it too hard for a beginner?

**Comment by alexcrichton** [comment]
This should be hopefully not too hard to implement! The parsing happens [around here](https://github.com/rustwasm/wasm-bindgen/blob/f5f541337cacfd09c997ce1a9b7623712d72a475/crates/macro-support/src/parser.rs#L241-L246) and we probably just want to store an enum of either an `Ident` or a `String`

**Comment by alexcrichton** [comment]
@brisad ah true! We probably will need to add a function that modifies the symbol name to avoid dollar-signs. It may be able to use the `ShortHash` type already there to just have a unique identifier at the end

**Comment by brisad** [comment]
@alexcrichton I'm attaching a WIP PR to get some comments on whether I'm on the right track or not. 

Instead of making it an enum like you said I just changed it to a `String` instead, as I couldn't see the use for `Ident` anymore. Let me know if that's a bad idea.

To modify the shim name, I simply filtered on ASCII alphanumeric chars to make it a valid identifier. That way the name will likely still be readable for debugging purposes (unless it is something with only special chars, like `$π`). As the data passed to `ShortHash` already contains the identifier for the Rust function, I figured we don't need to pass any more data to the hash function, as those identifiers should be unique.

But a big problem with the solution in the PR is that I now can do `#[wasm_bindgen(js_name="Some random text\n&*$#/")]` or `#[wasm_bindgen(js_name="alert('hi');foo"]`, and that will happily go right into the `.js` file. :-( Should we also validate that the passed name is a valid JS identifier?
